### ISSUE TITLE
selenium-server example fails to satisfy arguments dynamic fixture

### ISSUE THREAD

**Issue by jessesuen** [argoproj/argo#39]
When submitting the Selenium Demo template, the submission fails during preprocessing with the following error.

```
{
"code":"ERR_AX_ILLEGAL_ARGUMENT",
"message":"selenium-server 'inputs.parameters.BROWSER' was not satisfied by caller",
"detail":""
}
```

I suspect preprocessing is not handling argument substitution to dynamic fixtures correctly

**Comment by jessesuen** [comment]
The problem is during embedding. Here is a section of the embedded template for the selenium-server fixture:
```
{
  ...
  "selenium": {
    "template": {
      "type": "container",
      "version": "1",
      "name": "selenium-server",
      "description": "Base selenium container for selenium firefox and chrome",
      "id": "5437d66f-0c90-5245-5cd4-dd2ae7252368",
      "repo": "https://github.com/argoproj/appstore.git",
      "branch": "master",
      "revision": "b6441b788e8f9ae142238ab71f7fda21cfdd411f",
      "inputs": {
        "parameters": {
          "BROWSER": null
        }
      },
      "resources": {
        "mem_mib": "1024",
        "cpu_cores": "0.1"
      },
      "image": "selenium/standalone-%%inputs.parameters.BROWSER%%-debug:latest"
    },
    "status": 0,
    "status_string": "",
    "cost": 0,
    "create_time": 0,
    "launch_time": 0,
    "init_time": 0,
    "average_init_time": 0,
    "end_time": 0,
    "wait_time": 0,
    "run_time": 0,
    "average_runtime": 0,
    "artifact_tags": "",
    "is_submitted": false
  }
}
```
Notice that the container has `inputs.parameters.BROWSER`, but there is no `arguments` field (as a peer to `template`). There should be. The bug is is that during embedding of workflows, I neglect to apply the same inferring of template arguments that I do with workflow steps.

**Comment by jessesuen** [comment]
Fixed in 1.0.1 and 1.1

**Comment by jessesuen** [comment]
NOTE: after upgrading to the build with the fix, a cluster which had already attached the appstore repo, will need to disconnect and reconnect the repo to reingest the correctly embedded template.
